tests/main/snapd-reexec/task.yaml

```diff
@@ -94,6 +96,13 @@ execute: |
     fi
     if [ "$IS_REEXEC" = true ]; then
         /usr/bin/env SNAPD_DEBUG=1 snap list 2>&1 | MATCH "DEBUG: restarting into \"$SNAPD_MOUNT_DIR/current/usr/bin/snap\""        
+
+        if [ "${SNAPD_DEB_FROM_REPO-}" = "false" ]; then
+          echo "Ensure the command line has been updated in /proc"
+          snapd_pid="$(systemctl show -P MainPID snapd.service)"
+          snapd_cmdline="$(tr '\0' , <"/proc/${snapd_pid}/cmdline")"
+          test "${snapd_cmdline}" = "${SNAPD_MOUNT_DIR}/current/usr/lib/snapd/snapd," || exit 1
```

Comment:
What is the exit for, this should exit on its own due to `set -e`, right?

Reply:
Brain fart.